### PR TITLE
integration: use helmclient for teardown

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/4059.

Using helm client opens possibility to check if resource exists before deleting it.

### CODE REVIEW

File: integration/setup/setup.go [hunk 1]
```diff
@@ -76,8 +78,6 @@ func Setup(m *testing.M, config Config) {
 				// teardown errors are logged inside the function.
 				v = 1
 			}
-			// TODO there should be error handling for the framework teardown.
-			config.Host.Teardown()
```
Comment (kopiczko) [inline integration/setup/setup.go]: I moved it to `teardown`.

File: integration/setup/teardown.go [hunk 2]
```diff
@@ -46,6 +45,11 @@ func teardown(ctx context.Context, config Config) error {
 		}
 	}
 
+	{
+		// TODO there should be error handling for the framework teardown.
```
Comment (xh3b4sd) [inline integration/setup/teardown.go]: 100%

File: integration/setup/teardown.go [hunk 3]
```diff
@@ -30,7 +29,7 @@ func teardown(ctx context.Context, config Config) error {
 		}
 
 		for _, release := range releases {
-			err = framework.HelmCmd(fmt.Sprintf("delete %s --purge", release))
+			err = config.Resource.Delete(release)
```
Comment (xh3b4sd) [inline integration/setup/teardown.go]: This is nice.

**Comment by kopiczko** [comment]
> Nice, is there more to move though?

Sorry, I don't follow. What should be moved?

**Comment by xh3b4sd** [comment]
My question was if there is more to refactor in this regard because we do the helm command shell out also in a lot of other places. I wanted to know what is left and what the direction here is further down the road.

**Comment by kopiczko** [comment]
@xh3b4sd I updated the comment. Using API client opens possibility to check if release exists. I want to make `teardown` idempotent so we can call it before test execution.

**Comment by MarcelMue** [comment]
FYI: I am want to get the support in `resource` straight so we can use the client for basically everything we currently use `helmCmd` for. `InstallOperator` also uses `helmCmd` still under the hood.

**Pull request merged**